tmt/steps/provision/__init__.py

```diff
@@ -310,6 +315,17 @@ def _ssh_options(self, join=False):
         # Use the shared master connection
         options.append(f'-S{self._ssh_socket()}')
 
+        try:
+            # ssh cmdline grabs first cli option and ignores others.
+            # This way ve can override ssh default ssh options.
+            ssh_options = [v for opt in self.ssh_option for v in ("-o", opt)]
+            index = 0
+            for ssh_op in ssh_options:
+                options.insert(index, ssh_op)
+                index += 1
+        except TypeError:
+            pass
```

Comment:
* I don't understand what is `except TypeError` guarding against. What kind of issue may lead to `TypeError` being raised? 
* the `index += 1` smells like a job for `enumerate()`, but... IIUIC, loop adds new options to `options`, at the beginning. Wouldn't `options = ssh_options + options` do the exactly the same job?

Seems to me this could be simplified, given the default value of `self.ssh_option` to be `[]` AFAICT:

```python
# ssh cmdline grabs first cli option and ignores others.
# This way ve can override ssh default ssh options.
options = [v for opt in self.ssh_option for v in ("-o", opt)] + options

# Or, since f-strings are a thing and nested comprehensions harder to follow:
options = [f'-o{opt}' for opt in self.ssh_option] + options
```

* no-op when `self.ssh_option` is empty,
* no need for `insert()` or extra variables,
* follows the `-oFoo` approach already stored in `options`.

Reply:
Thanks. Changed.